### PR TITLE
sys/net/nanocoap: introduce Accept option helper

### PR DESCRIPTION
**Pull request by JKRhb**
### Contribution description

Working with content negotiation in the context of gcoap/nanocoap, I noticed that it would be handy to be able to access the Accept option of a request more easily. Therefore, this PR adds a new helper function `coap_get_accept` that gets the respective option value from a CoAP packet. It reuses most of the code from the existing `coap_get_content_type` helper, moving its logic to a new internal function that is called by both `coap_get_content_type` and `coap_get_accept`.

### Testing procedure

I adjusted the existing unit tests to assert that new the helper works.

### Issues/PRs references

—

### CODE REVIEW

File: sys/net/application_layer/nanocoap/nanocoap.c [hunk 1]
```diff
@@ -280,7 +280,7 @@ uint8_t *coap_iterate_option(coap_pkt_t *pkt, uint8_t **optpos,
     }
 }
 
-unsigned coap_get_content_type(coap_pkt_t *pkt)
+static unsigned _get_content_format(coap_pkt_t *pkt, unsigned int opt_num)
 {
     uint8_t *opt_pos = coap_find_option(pkt, COAP_OPT_CONTENT_FORMAT);
```
Comment (miri64) [inline sys/net/application_layer/nanocoap/nanocoap.c:285]: This should be
```suggestion
    uint8_t *opt_pos = coap_find_option(pkt, opt);
```

no?

Comment (JKRhb) [reply sys/net/application_layer/nanocoap/nanocoap.c]: You are right, thanks :)

File: tests/unittests/tests-nanocoap/tests-nanocoap.c [hunk 2]
```diff
@@ -155,6 +155,10 @@ static void test_nanocoap__put_req(void)
     TEST_ASSERT_EQUAL_INT(fmt_opt_len, len);
     TEST_ASSERT_EQUAL_INT(COAP_FORMAT_TEXT, coap_get_content_type(&pkt));
 
+    len = coap_opt_add_uint(&pkt, COAP_OPT_ACCEPT, COAP_FORMAT_TEXT);
+    TEST_ASSERT_EQUAL_INT(fmt_opt_len, len);
+    TEST_ASSERT_EQUAL_INT(COAP_FORMAT_TEXT, coap_get_accept(&pkt));
+
```
[the remark this answers is not in the document]
Comment (JKRhb) [reply tests/unittests/tests-nanocoap/tests-nanocoap.c]: Good point ;) I replaced `COAP_FORMAT_TEXT` with `COAP_FORMAT_CBOR` in 977f8fa.

File: sys/net/application_layer/nanocoap/nanocoap.c (hunk 1 above)
Comment (miri64) [reply sys/net/application_layer/nanocoap/nanocoap.c]: Should have been `opt_num` :sweat_smile:

Comment (JKRhb) [reply sys/net/application_layer/nanocoap/nanocoap.c]: Oh, sorry, I just hit the commit button :sweat_smile: Should now be the correct variable name :)

Comment (miri64) [reply sys/net/application_layer/nanocoap/nanocoap.c]: Also sorry for the mistake in the first place ^^"

File: tests/unittests/tests-nanocoap/tests-nanocoap.c [hunk 3]
```diff
@@ -155,6 +155,10 @@ static void test_nanocoap__put_req(void)
     TEST_ASSERT_EQUAL_INT(fmt_opt_len, len);
     TEST_ASSERT_EQUAL_INT(COAP_FORMAT_TEXT, coap_get_content_type(&pkt));
 
+    len = coap_opt_add_uint(&pkt, COAP_OPT_ACCEPT, COAP_FORMAT_CBOR);
+    TEST_ASSERT_EQUAL_INT(fmt_opt_len, len);
```
Comment (miri64) [inline tests/unittests/tests-nanocoap/tests-nanocoap.c:159]: Seems, due to CBOR being chosen, the length of that option is longer now longer than 1.

```suggestion
    size_t accept_opt_len = 2;
    len = coap_opt_add_uint(&pkt, COAP_OPT_ACCEPT, COAP_FORMAT_CBOR);
    TEST_ASSERT_EQUAL_INT(accept_opt_len, len);
```

I would prefer the variable declaration to be moved up top, but the Github interface makes it a bit difficult to suggest this, so please adapt manually.

File: tests/unittests/tests-nanocoap/tests-nanocoap.c [hunk 4]
```diff
@@ -155,6 +155,10 @@ static void test_nanocoap__put_req(void)
...
+    TEST_ASSERT_EQUAL_INT(COAP_FORMAT_CBOR, coap_get_accept(&pkt));
+
     len = coap_opt_finish(&pkt, COAP_OPT_FINISH_PAYLOAD);
     TEST_ASSERT_EQUAL_INT(total_hdr_len + uri_opt_len + fmt_opt_len + 1, len);
```
Comment (miri64) [inline tests/unittests/tests-nanocoap/tests-nanocoap.c:163]: Also, this check needs to be amended:
```suggestion
    TEST_ASSERT_EQUAL_INT(
        total_hdr_len + uri_opt_len + fmt_opt_len + accept_opt_len + 1, len
    );
```

**Review by miri64 — CHANGES_REQUESTED** [review]

**Review by miri64 — APPROVED** [review]
Please squash!

**Pull request merged**